diff --git a/target/linux/ramips/dts/mt7621.dtsi b/target/linux/ramips/dts/mt7621.dtsi
index 3e076741a9..375f7f97ec 100644
--- a/target/linux/ramips/dts/mt7621.dtsi
+++ b/target/linux/ramips/dts/mt7621.dtsi
@@ -436,7 +436,15 @@
                clocks = <&nficlock>;
                clock-names = "nfi_clk";
        };
+        crypto: crypto@1e004000 {
+         status = "okay";

+         compatible = "mediatek,mtk-eip93";
+         reg = <0x1e004000 0x1000>;
+
+         interrupt-parent = <&gic>;
+         interrupts = <GIC_SHARED 19 IRQ_TYPE_LEVEL_HIGH>;
+        };
        ethernet: ethernet@1e100000 {
                compatible = "mediatek,mt7621-eth";
                reg = <0x1e100000 0x10000>;
